.github/actions/build-cache-key/action.yml

```diff
@@ -8,6 +8,9 @@ inputs:
     description: 'String prefix applied to the build cache key'
     required: false
     default: 'build'
+  paths:
```

Comment:
> taking a step back, as a generic cache key, it's not necessary composed of file hashes. what if we just omit `kong/**` from build-cache-key workflow, and in release.yml we will just
> 
> ```
> key: ${{ steps.cache-key.outputs.cache-key }}-${{ hashFiles('kong/**') }}
> ```

This is a good point. The gripe I still have with this is that it encourages more copy/paste, and sooner or later somebody will see `hashFiles()` in there and take this as their cue to add another file/glob. Eventually we're back to where we started with a super long line that is hard to read and painful to fix when there are merge conflicts:

```yaml
key: ${{ steps.cache-key.outputs.cache-key }}-${{ hashFiles('kong/**', 'kong-extra/**', 'my-build-file.ext', 'my-other-directory/*/*.txt') }}
```

Callers of the action _could_ achieve the same by adding `hashFiles()` to the `prefix` input option:

```
- name: Generate cache key
  id: cache-key
  uses: ./.github/actions/build-cache-key
  with:
    prefix: build::${{ hashFiles('kong/**') }}
```

It's somewhat better, but we still have the problem where new paths are added, and eventually it becomes a super long line.

Reply:
is it possible to just call `hashFiles` in `Generate cache key`? I'm mostly concerned about the fact that we are implementing our own method of hash and it seems bit complex.
something like `sha256sum ( ${{ hashFiles(preset path1) }}-${{ hashFiles(preset path2) }}-${{ hashFiles(custom path1) }} )`

Reply:
> I'm mostly concerned about the fact that we are implementing our own method of hash and it seems bit complex.

The primary reason for not using `hashFiles()` is that GHA's expression language doesn't really support non-primitive inputs. Otherwise we could do something like:

```yaml
- name: Generate cache key
  id: cache-key
  uses: ./.github/actions/build-cache-key
  with:
    paths:
      - kong/**
      - foo.txt
```

...and then call `hashFiles(inputs.paths)` somewheres.

Reply:
how about something like
```
- name: Generate cache key
  id: cache-key
  uses: ./.github/actions/build-cache-key
  with:
    suffixes:
      - ${{ hashFiles('kong/**') }}
      - ${{ hashFiles('foo.txt') }}
```

then we can even have

```
- name: Generate cache key
  id: cache-key
  uses: ./.github/actions/build-cache-key
  with:
    suffixes:
    - fips
```

i know we want to make this easy to use, but the cost of maintaining a new hashFiles implementation is also what we need to consider.

Reply:
The problem is that action inputs _cannot_ be arrays though, only primitives: string, boolean, number.

Check out the latest iteration and let me know what you think.

Reply:
i see, in that case, do we still need to compute filehashs and sort them?
```
        FILE_HASHES=(
          ${{ hashFiles('.bazelignore') }}
          ${{ hashFiles('.bazelrc') }}
          ${{ hashFiles('.bazelversion') }}
          ${{ hashFiles('.github/actions/build-cache-key/**') }}
          ${{ hashFiles('.github/workflows/build.yml') }}
          ${{ hashFiles('.requirements') }}
          ${{ hashFiles('BUILD.bazel') }}
          ${{ hashFiles('WORKSPACE') }}
          ${{ hashFiles('bin/kong') }}
          ${{ hashFiles('bin/kong-health') }}
          ${{ hashFiles('build/**') }}
          ${{ hashFiles('kong-*.rockspec') }}
          ${{ hashFiles('kong.conf.default') }}
          ${{ hashFiles('kong/**') }}
        )
```
could be a single
`${{ hashFiles('.bazelignore', '.bazelrc', ...)) }}` and we don't need all those bash logic, so only
`sha256sum(hashFiles("default files") + "." + inputs.extra))`

Reply:
> I could rewrite this in javascript and get more readable array/string-munging code, but I think I'd need to pull in a crypto dependency for the hash bit
Probably we will just go with perl 😄 
anyway, i think it's now in good shape. is there anything left before transitioning to non-draft?